R/utils_render_rtf_redo.R

```diff
@@ -240,6 +240,134 @@ rtf_table <- function(rows) {
   rtf_raw(paste(unlist(rows), collapse = ""))
 }
 
+# This is the assumed width of the area within page
+# margins in a default Word document
+standard_width_twips <- 9468L
+
+twip_factors <-
+  c(
+    `in` = 1440, `pt` = 20, `px` = 15,
+    `cm` = 566.9291, `mm` = 56.69291, `tw` = 1
+  )
+
+parse_length_str <- function(lengths_vec) {
+
+  if (length(lengths_vec) == 1 && lengths_vec == "auto") {
+    lengths_vec <- paste0(standard_width_twips, "tw")
+  }
+
+  match <- regexec("^([0-9.-]+)(%|[a-z]+)$", lengths_vec)
+  match <- regmatches(lengths_vec, match)
+
+  vals <- vapply(match, function(m) {
+    if (length(m)) {
+      as.numeric(m[[2]])
+    } else {
+      NA_real_
+    }
+  }, numeric(1))
+
+  units <- vapply(match, function(m) {
+    if (length(m)) {
+      m[[3]]
+    } else {
+      NA_character_
+    }
+  }, character(1))
+
+  bad_values <- nzchar(lengths_vec) & is.na(vals)
+
+  if (any(bad_values)) {
+    stop(
+      "Some of the values supplied as absolute lengths cannot be interpreted:\n",
+      "* Use either of: `px`, `pt`, `in`, `cm`, `mm`, or `tw`.",
```

Comment:
Let's include the bad values in the error message

Reply:
Now implemented.